src/maps.rs

```diff
@@ -182,31 +245,30 @@ pub(crate) fn parse(pid: Pid) -> Result<impl Iterator<Item = Result<MapsEntry>>>
     Ok(iter)
 }
 
-/// A helper function checking whether a `MapsEntry` has relevant to
-/// symbolization efforts. If that is not the case, it may be possible to ignore
-/// it altogether.
-pub(crate) fn is_symbolization_relevant(entry: &MapsEntry) -> bool {
-    // Only entries with actual paths are of relevance.
-    if entry.path.as_path().components().next() != Some(Component::RootDir) {
-        return false
-    }
+/// A helper function checking whether a `MapsEntry` has relevance to
+/// symbolization efforts and converting it accordingly.
+pub(crate) fn check_symbolization_relevant(entry: MapsEntry) -> Option<PathMapsEntry> {
+    let MapsEntry {
+        range,
+        mode,
+        offset,
+        path_name,
+    } = entry;
 
     // Only entries that are executable and readable (r-x-) are of relevance.
-    if (entry.mode & 0b1010) != 0b1010 {
-        return false
+    if (mode & 0b1010) != 0b1010 {
```

Comment:
is there a reason to dictate that mapping has to be only readable? I don't know if that actually matters in practice, but the less assumptions we make, the better. So maybe let's just check that executable bit is set?

Reply:
Yeah, I thought about that, too. I can adjust.